web_external/js/views/body/UploadDatasetView.js

```diff
@@ -257,10 +255,61 @@ isic.views.UploadDatasetView = isic.View.extend({
     }
 });
 
+isic.views.UploadDatasetRequestView = isic.View.extend({
+    events: {
+        'submit #isic-dataset-form': function (event) {
+            event.preventDefault();
+            this.$('#isic-dataset-submit').prop('disabled', true);
+            this.requestCreateDatasetPermission();
+        }
+    },
+
+    initialize: function (settings) {
+        this.render();
+    },
+
+    render: function () {
+        this.$el.html(isic.templates.uploadDatasetRequest({
+            user: girder.currentUser
+        }));
+
+        return this;
+    },
+
+    requestCreateDatasetPermission: function () {
+        girder.restRequest({
+            path: 'user/requestCreateDatasetPermission',
+            type: 'POST'
+        }).done(_.bind(function (resp) {
+            if (_.has(resp, 'extra') && resp.extra === 'hasPermission') {
+                // Directly update user permissions
+                girder.currentUser.setCanCreateDataset(true);
```

Comment:
Exactly. Perhaps it's not good, because this will only happen if they click the Request Access button again, but I wanted some solution for the user to discover they have permission short of refreshing the app or logging out. The other solution is to revert to updating the user on the client in the upload dataset route.